feat(ui): disable GSAP-based animations for IE and touch enabled devices

milesap opened the pull request:
## Description
- Disabled all GSAP animations for IE, and most for touch devices (when `$rootElement` was available)
- Does not remove GSAP animation, but rather immediately moves the timeline to completion 

Closes #1465

## Testing
:eye: balling in IE & chrome

## Documentation
inline where needed

## Checklist
<!-- Quick checklist for items that are easy to miss -->

- [x] commits messages follow the guidelines
- [x] code passes unit tests
- [x] release notes have been updated
- [x] PR targets the correct release version

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/fgpv-vpgf/fgpv-vpgf/1556)
<!-- Reviewable:end -->

milesap commented:
Review status: 0 of 9 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[src/app/geo/overview-toggle.directive.js, line 123 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1556#-K_euyJBjldG8MS33jit:-K_eyR9DDcIU-78_Zha4:b-m0pd6g) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/c7e11b212cffc6bb2b37c6ae221b70e94dbeea6e/src/app/geo/overview-toggle.directive.js#L123)):*
<details><summary><i>Previously, AleksueiR (Aleksuei Riabtsev) wrote…</i></summary><blockquote>

Wouldn't `exportRoot` affect __all__ timelines at the same time? We need to just skip the current animation, not all of them. This might work, but I don't think it's the right way to do it. Also, setting a progress to 1 on a reversed timeline will move the playhead to the end, not the beginning.

Also, `RV.isIE || $rootElement.hasClass('rv-touch')` is repeated several times. Maybe it makes sense to have two overloaded service functions: `play` and `reverse` to handle that.
```js

$rootScope.overviewActive ? animationService.reverse(overviewAnimation) : animationService.play(overviewAnimation);

///

function play(timeline) {
         timeline.play();    

      if (RV.isIE || RV.isTouch) {
         timeline.progress(1);
    }
}

function reverse(timeline) {
         timeline.reverse();    

      if (RV.isIE || RV.isTouch) {
         timeline.progress(0);
    }
}
```
</blockquote></details>

Correct, `exportRoot` affects only the timelines created prior to it. This is why they are placed immediately following a timeline animation. Regardless, since we are also interested in disabling all animations anyways this shouldn't be an issue.

`progress(1)` moves the timeline to completion, not the start or end. A reversing animation with `progress(1)` will move to the start, while a forward animation would move to the end. Here's a [codepen example](https://codepen.io/anon/pen/MJYawL) of this in action (I added the progress button).

Would you suggest a separate service with `play` and `reverse` be create? I'm OK with this - though 1 or 2 files may not have access to this service, see below.

---

*[src/app/ui/common/plug-slide.animation.js, line 298 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1556#-K_evf5VO_FafdBW0Dy7:-K_f07tRCv9l0OayXthg:b-2snbb8) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/c7e11b212cffc6bb2b37c6ae221b70e94dbeea6e/src/app/ui/common/plug-slide.animation.js#L298)):*
<details><summary><i>Previously, AleksueiR (Aleksuei Riabtsev) wrote…</i></summary><blockquote>

Why only in IE, what about touch?
</blockquote></details>

I wasn't sure if dependency injection is available here - if so how would this be done?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1556)*
<!-- Sent from Reviewable.io -->

AleksueiR commented:
Review status: 0 of 9 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[src/app/geo/overview-toggle.directive.js, line 123 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1556#-K_euyJBjldG8MS33jit:-K_ibyuajzIAQDax9LIg:b-cl9hes) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/c7e11b212cffc6bb2b37c6ae221b70e94dbeea6e/src/app/geo/overview-toggle.directive.js#L123)):*
<details><summary><i>Previously, milesap (Miles Petrov) wrote…</i></summary><blockquote>

Correct, `exportRoot` affects only the timelines created prior to it. This is why they are placed immediately following a timeline animation. Regardless, since we are also interested in disabling all animations anyways this shouldn't be an issue.

`progress(1)` moves the timeline to completion, not the start or end. A reversing animation with `progress(1)` will move to the start, while a forward animation would move to the end. Here's a [codepen example](https://codepen.io/anon/pen/MJYawL) of this in action (I added the progress button).

Would you suggest a separate service with `play` and `reverse` be create? I'm OK with this - though 1 or 2 files may not have access to this service, see below.
</blockquote></details>

Strange, if you look at [this demo](https://greensock.com/timelinelite), the progress decreases to 0 when the animation is reversed. 

I think have a separate service for `play/reverse` is better as `exportRoot` can potentially have unpredictable results. 

---

*[src/app/ui/common/plug-slide.animation.js, line 61 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1556#-K_ibhCqI3Uqtum6-vgW:-K_ibhCqI3Uqtum6-vgX:bamylee) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/c7e11b212cffc6bb2b37c6ae221b70e94dbeea6e/src/app/ui/common/plug-slide.animation.js#L61)):*
> ```JavaScript
>      */
>     function animationBuilder(type, direction, grand) {
>         return $rootElement => {
> ```

Here we inject `$rootElement` into the animation function built by a builder function.

---

*[src/app/ui/common/plug-slide.animation.js, line 298 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1556#-K_evf5VO_FafdBW0Dy7:-K_ibd88gf-R4R2_nwxO:b-r05h4p) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/c7e11b212cffc6bb2b37c6ae221b70e94dbeea6e/src/app/ui/common/plug-slide.animation.js#L298)):*
<details><summary><i>Previously, milesap (Miles Petrov) wrote…</i></summary><blockquote>

I wasn't sure if dependency injection is available here - if so how would this be done?
</blockquote></details>

It is possible. See here: https://reviewable.io/reviews/fgpv-vpgf/fgpv-vpgf/1556#-K_ibhCqI3Uqtum6-vgW

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1556)*
<!-- Sent from Reviewable.io -->

milesap commented:
Review status: 0 of 12 files reviewed at latest revision, 3 unresolved discussions.

---

*[src/app/geo/overview-toggle.directive.js, line 123 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1556#-K_euyJBjldG8MS33jit:-KaDdbGj2jI-JB0d12ZS:bc7a6wz) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/c7e11b212cffc6bb2b37c6ae221b70e94dbeea6e/src/app/geo/overview-toggle.directive.js#L123)):*
<details><summary><i>Previously, AleksueiR (Aleksuei Riabtsev) wrote…</i></summary><blockquote>

Strange, if you look at [this demo](https://greensock.com/timelinelite), the progress decreases to 0 when the animation is reversed. 

I think have a separate service for `play/reverse` is better as `exportRoot` can potentially have unpredictable results. 
</blockquote></details>

`animationService` has custom `to` and `fromTo` methods which shorten the animation.

---

*[src/app/ui/common/plug-slide.animation.js, line 61 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1556#-K_ibhCqI3Uqtum6-vgW:-KaDeBrJul7FscbBlbWZ:b-cv6zbk) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/c7e11b212cffc6bb2b37c6ae221b70e94dbeea6e/src/app/ui/common/plug-slide.animation.js#L61)):*
<details><summary><i>Previously, AleksueiR (Aleksuei Riabtsev) wrote…</i></summary><blockquote>

Here we inject `$rootElement` into the animation function built by a builder function.
</blockquote></details>

All fixed

---

*[src/app/ui/common/plug-slide.animation.js, line 298 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1556#-K_evf5VO_FafdBW0Dy7:-KaDeDGcmN633uSlBLxr:b-cv6zbk) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/c7e11b212cffc6bb2b37c6ae221b70e94dbeea6e/src/app/ui/common/plug-slide.animation.js#L298)):*
<details><summary><i>Previously, AleksueiR (Aleksuei Riabtsev) wrote…</i></summary><blockquote>

It is possible. See here: https://reviewable.io/reviews/fgpv-vpgf/fgpv-vpgf/1556#-K_ibhCqI3Uqtum6-vgW
</blockquote></details>

All fixed

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1556)*
<!-- Sent from Reviewable.io -->

AleksueiR commented:
Reviewed 11 of 12 files at r2.
Review status: 11 of 12 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[src/app/geo/overview-toggle.directive.js, line 123 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1556#-K_euyJBjldG8MS33jit:-KaHaR3qUR3siSN4mCU7:bim7ilc) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/c7e11b212cffc6bb2b37c6ae221b70e94dbeea6e/src/app/geo/overview-toggle.directive.js#L123)):*
<details><summary><i>Previously, milesap (Miles Petrov) wrote…</i></summary><blockquote>

`animationService` has custom `to` and `fromTo` methods which shorten the animation.
</blockquote></details>

Why overloading `to` and `fromTo` instead of `play/reverse`? This seems more complicated to me. Also, there is a `from` call which also needs to be overloaded if you are doing it this way.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1556)*
<!-- Sent from Reviewable.io -->

milesap commented:
Review status: 11 of 12 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[src/app/geo/overview-toggle.directive.js, line 123 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1556#-K_euyJBjldG8MS33jit:-KaHbwCaCjLDqzau36rz:b-fo7loa) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/c7e11b212cffc6bb2b37c6ae221b70e94dbeea6e/src/app/geo/overview-toggle.directive.js#L123)):*
<details><summary><i>Previously, AleksueiR (Aleksuei Riabtsev) wrote…</i></summary><blockquote>

Why overloading `to` and `fromTo` instead of `play/reverse`? This seems more complicated to me. Also, there is a `from` call which also needs to be overloaded if you are doing it this way.
</blockquote></details>

Went this route since `TweenLite` has static methods `to`, `fromTo`, and `from` that are played immediately and there are a bunch of animation methods for `TimelineLite` such as `play`, `reverse`, `reversed` etc. Forgot the `from` method, I'll push an update soon.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1556)*
<!-- Sent from Reviewable.io -->

AleksueiR commented:
Review status: 11 of 12 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[src/app/geo/overview-toggle.directive.js, line 123 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1556#-K_euyJBjldG8MS33jit:-KaHdVD3CKAPRnVvIg94:b-gxztsp) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/c7e11b212cffc6bb2b37c6ae221b70e94dbeea6e/src/app/geo/overview-toggle.directive.js#L123)):*
<details><summary><i>Previously, milesap (Miles Petrov) wrote…</i></summary><blockquote>

Went this route since `TweenLite` has static methods `to`, `fromTo`, and `from` that are played immediately and there are a bunch of animation methods for `TimelineLite` such as `play`, `reverse`, `reversed` etc. Forgot the `from` method, I'll push an update soon.
</blockquote></details>

Fair enough.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1556)*
<!-- Sent from Reviewable.io -->

AleksueiR commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> Nice optimizations :+1: 

---

Reviewed 2 of 2 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1556#-:-KaMpTBRprYNESVwxBD-:bdhz9g8)*
<!-- Sent from Reviewable.io -->

alyec commented:
Reviewed 10 of 12 files at r2, 2 of 2 files at r3.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[src/app/layout/animation.service.js, line 48 at r3](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1556#-KaN4vRVSdfsvy5vgvoz:-KaN4vRVSdfsvy5vgvp-:b-fttb37) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/6a99e8f283b52bd230e48886f1c5a52e4af23a6a/src/app/layout/animation.service.js#L48)):*
> ```JavaScript
>         * @param     {String}   toWrap          the name of the existing method in TLLinstance to be wrapped
>         * @param     {Object}   TLLinstance     an instance of TimelineLite, defaults to static TweenLite class
>         * @return    {Function} a function which effectivly replaces the original toWrap function
> ```

typo effectively

---

*[src/app/layout/animation.service.js, line 53 at r3](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1556#-KaN544LgkU2_QzjLqxV:-KaN544LgkU2_QzjLqxW:b-26zra1) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/6a99e8f283b52bd230e48886f1c5a52e4af23a6a/src/app/layout/animation.service.js#L53)):*
> ```JavaScript
>             return function () {
>                 const args = [...arguments];
>                 // set duration to 0.001 when animations are disabled so they complete almost immediatly. Cannot be 0 or it fails.
> ```

typo immediately

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1556)*
<!-- Sent from Reviewable.io -->

milesap commented:
Review status: 11 of 12 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[src/app/layout/animation.service.js, line 48 at r3](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1556#-KaN4vRVSdfsvy5vgvoz:-KaNBRUx3XeJGrESlb8_:b-886gsx) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/6a99e8f283b52bd230e48886f1c5a52e4af23a6a/src/app/layout/animation.service.js#L48)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary><blockquote>

typo effectively
</blockquote></details>

Fixed

---

*[src/app/layout/animation.service.js, line 53 at r3](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1556#-KaN544LgkU2_QzjLqxV:-KaNBPSjwcxJQM5376oR:b-886gsx) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/6a99e8f283b52bd230e48886f1c5a52e4af23a6a/src/app/layout/animation.service.js#L53)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary><blockquote>

typo immediately
</blockquote></details>

Fixed

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1556)*
<!-- Sent from Reviewable.io -->

alyec commented:
Reviewed 1 of 1 files at r4.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1556)*
<!-- Sent from Reviewable.io -->

AleksueiR commented:
Reviewed 1 of 1 files at r4.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1556)*
<!-- Sent from Reviewable.io -->

AleksueiR commented:
Reviewed 1 of 1 files at r5.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1556)*
<!-- Sent from Reviewable.io -->
